ui/apps/platform/src/Containers/NetworkGraph/deployment/ExternalFlows.tsx

```diff
@@ -164,169 +201,227 @@ function ExternalFlows({
+                                <FlowTable
+                                    pagination={baseline.urlPagination}
+                                    flowCount={totalBaseline}
+                                    statusType="BASELINE"
+                                    tableState={baseline.tableState}
+                                    areAllRowsSelected={areAllPageBaselineSelected}
+                                    urlSearchFiltering={urlSearchFiltering}
+                                    onSelectAll={selectAllBaselineFlows}
+                                    isFlowSelected={isFlowSelected}
+                                    onRowSelect={onSelectFlow}
+                                    rowActions={(flow) => [
+                                        {
+                                            title: <span>Mark as anomalous</span>,
+                                            onClick: async (e) => {
+                                                e.preventDefault();
+                                                confirmStatusChange([flow], 'ANOMALOUS');
+                                            },
                                         },
-                                    },
-                                ]}
-                            />
-                        </ExpandableSection>
-                    </StackItem>
-                </Stack>
-            </StackItem>
-        </Stack>
+                                    ]}
+                                />
+                            </ExpandableSection>
+                        </StackItem>
+                    </Stack>
+                </StackItem>
+            </Stack>
+            <ConfirmationModal
+                title="Apply status change to multiple flows?"
+                ariaLabel="Confirm status change"
+                confirmText="Apply"
+                isOpen={isConfirmingStatusChange}
+                onConfirm={onConfirmStatusChange}
+                onCancel={onCancelStatusChange}
+                isDestructive={false}
+            >
+                <p>
+                    All flows that have the same combination of direction, port, and protocol have
+                    the same status. This action will affect the status of all matching flows, even
+                    flows that you did not select.
+                </p>
+                <div style={{ maxHeight: '300px', overflowY: 'auto' }}>
+                    <Table variant="compact" borders={false} isStickyHeader>
+                        <Thead>
+                            <Tr>
+                                <Th>Direction</Th>
+                                <Th>Port / protocol</Th>
+                            </Tr>
+                        </Thead>
+                        <Tbody>
+                            {uniquePendingFlows.map(({ direction, port, protocol }) => (
+                                <Tr key={`${direction}-${port}-${protocol}`}>
```

Comment:
Delayed realization that function pre-computed this value as `key` prop of object.